source/beamline/SGM/energy/SGMUndulatorControl.h

```diff
@@ -0,0 +1,87 @@
+#ifndef SGMUNDULATORCONTROL_H
+#define SGMUNDULATORCONTROL_H
+
+#define DEFAULT_UNDULATOR_VELOCITY 5000
+#define DEFAULT_UNDULATOR_ACCELERATION  2500
+
+#include "beamline/AMPseudoMotorControl.h"
+
+class SGMUndulatorControl : public AMPseudoMotorControl
+{
+	Q_OBJECT
+public:
+	explicit SGMUndulatorControl(QObject *parent = 0);
```

Comment:
It looks like the `SGMUndulatorControl` and the `SGMGratingControl` are almost identical.  As a future cleanup, I'm pretty sure they could inherit from a single class that does all the work.